### PR TITLE
[CHIA-1308] Port wallet node RPC endpoints to `@marshal` decorator

### PR DESCRIPTION
**Pull request by Quexington**
This is another PR in the same vein as https://github.com/Chia-Network/chia-blockchain/pull/18593 .

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -694,20 +693,23 @@ async def push_transactions(
 
                 interface.side_effects.transactions.extend(inner_action_scope.side_effects.transactions)
 
-        return {}
+        return PushTransactionsResponse([], [])  # tx_endpoint takes care of this
```
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:696]: seems fishy to need this if it's pointless?

File: chia/_tests/wallet/rpc/test_wallet_rpc.py [hunk 2]
```diff
@@ -405,29 +412,25 @@ async def test_push_transactions(wallet_rpc_environment: WalletRpcTestEnvironmen
...
+    resp_client = await client.push_transactions(
+        PushTransactions(transactions=[tx], fee=uint64(10)),  # pylint: disable=unexpected-keyword-arg
+        DEFAULT_TX_CONFIG,
+    )
     resp = await client.fetch(
         "push_transactions", {"transactions": [tx.to_json_dict_convenience(wallet_node.config)], "fee": 10}
     )
     assert resp["success"]
-    resp = await client.fetch("push_transactions", {"transactions": [tx.to_json_dict()], "fee": 10})
+    resp = await client.fetch("push_transactions", {"transactions": [bytes(tx).hex()], "fee": 10})
```
Comment (altendky) [inline chia/_tests/wallet/rpc/test_wallet_rpc.py:423]: are we changing the api?  or did `tx.to_json_dict()` create a string before?

File: chia/_tests/wallet/rpc/test_wallet_rpc.py [hunk 3]
```diff
@@ -513,8 +518,8 @@ async def test_get_timestamp_for_height(wallet_rpc_environment: WalletRpcTestEnv
 
     await generate_funds(full_node_api, env.wallet_1)
 
-    # This tests that the client returns a uint64, rather than raising or returning something unexpected
-    uint64(await client.get_timestamp_for_height(uint32(1)))
+    # This tests that the client returns a sucessfully, rather than raising or returning something unexpected
```
Comment (altendky) [inline chia/_tests/wallet/rpc/test_wallet_rpc.py:521]:
```suggestion
    # This tests that the client returns successfully, rather than raising or returning something unexpected
```

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/cmds/signer.py [hunk 4]
```diff
@@ -292,7 +298,10 @@ class PushTransactionsCMD:
 
     async def run(self) -> None:
         async with self.rpc_info.wallet_rpc() as wallet_rpc:
-            await wallet_rpc.client.push_transactions(self.txs_in.transaction_bundle.txs)
+            # TODO: no default here
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/cmds/signer.py:301]: The command needs a bunch of common arguments that it doesn't currently have to allow the tx config to be configurable.  Besides being out of scope, I've done a lot of that work in a different PR: https://github.com/Chia-Network/chia-blockchain/pull/18641 and didn't wish to duplicate the effort.  The endpoint as it was kind of initially designed wasn't taking into account that additional transactions might be generated, so this was sort of implicit already and now it's more visible.

File: chia/rpc/wallet_rpc_api.py (hunk 1 above)
Comment (Quexington) [reply chia/rpc/wallet_rpc_api.py:696]: Agreed, there's a tension between the two wrappers `@marshal` and `@tx_endpoint`.  `@tx_endpoint` is responsible for filling in the common `transactions` and `unsigned_transactions` arguments, but it does this at its own layer where it thinks the response is a dict (because many RPCs don't yet use `@marshal` so it is a dict there) and therefore we have to return a silly object here in order to fit the `@marshal` paradigm which requires addressing these arguments.  On the client side `PushTransactionsResponse` will have this information filled in and we could in theory fill it in in the endpoint here but it would be pointless as `@tx_endpoint` is just going to override it.

TLDR: While we sit between paradigms, this is some unfortunately necessary silliness.  It's consistent with other places in the RPC and therefore can be easily targeted in a future refactor.

File: chia/_tests/wallet/rpc/test_wallet_rpc.py (hunk 2 above)
Comment (Quexington) [reply chia/_tests/wallet/rpc/test_wallet_rpc.py:423]: Actually, this is a test coverage change.  The client used to send bytes across and the two lines below the first added coverage to when it sends the two types of dicts across.  I changed the client to start sending dicts across so now I changed the line that was adding coverage for the dicts to instead add coverage for the bytes.

File: chia/cmds/signer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (altendky) [reply chia/cmds/signer.py:301]: ?

```suggestion
            # TODO: provide access to additional parameters instead of filling with the defaults constant
```

File: chia/_tests/wallet/rpc/test_wallet_rpc.py (hunk 2 above)
Comment (altendky) [reply chia/_tests/wallet/rpc/test_wallet_rpc.py:423]: So the `transactions` field already was a union effectively and remains unchanged in what is properly handled there?

Comment (Quexington) [reply chia/_tests/wallet/rpc/test_wallet_rpc.py:423]: Correct

**Review by altendky — APPROVED** [review]

**Review by markelrod — APPROVED** [review]
